File: app/src/main/java/net/squanchy/eventdetails/widget/EventDetailsHeaderLayout.java
package net.squanchy.eventdetails.widget;

import android.content.Context;
import android.support.design.widget.AppBarLayout;
import android.util.AttributeSet;
import android.widget.TextView;

import net.squanchy.R;
import net.squanchy.schedule.domain.view.Event;
import net.squanchy.support.widget.SpeakerView;

public class EventDetailsHeaderLayout extends AppBarLayout {

    private TextView titleView;
    private SpeakerView speakerView;

    public EventDetailsHeaderLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();

        titleView = (TextView) findViewById(R.id.title);
        speakerView = (SpeakerView) findViewById(R.id.speaker_container);
    }

    void updateWith(Event event, SpeakerView.OnSpeakerClickListener listener) {
        titleView.setText(event.title());
        titleView.setVisibility(VISIBLE);

        speakerView.setVisibility(event.speakers().isEmpty() ? GONE : VISIBLE);
        speakerView.updateWith(event.speakers(), listener);
    }
}

File: app/src/main/java/net/squanchy/eventdetails/widget/EventDetailsHeaderLayout.kt
package net.squanchy.eventdetails.widget

import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.core.view.isVisible
import com.google.android.material.appbar.AppBarLayout
import kotlinx.android.synthetic.main.activity_event_details.view.*
import net.squanchy.schedule.domain.view.Event
import net.squanchy.support.widget.SpeakerView

class EventDetailsHeaderLayout(context: Context, attrs: AttributeSet) : AppBarLayout(context, attrs) {

    internal fun updateWith(event: Event, listener: SpeakerView.OnSpeakerClickListener) {
        titleTextView.text = event.title
        titleTextView.isVisible = true

        speakerDetailsView.visibility = if (event.speakers.isEmpty()) View.GONE else View.VISIBLE
        speakerDetailsView.updateWith(event.speakers, listener)
    }
}
